auspicePaths.js

```diff
@@ -45,4 +56,85 @@ const auspicePaths = [
   "/fetch/*"
 ];
 
-module.exports = auspicePaths;
+/**
+ * Express middleware to examine a path and determine if it should be routed to Auspice
+ * for visualization, or sent a Gatsby-generated page.
+ * This may require a fetch call for the main dataset (or the meta.json if v1 schema)
+ * to decide if a URL is an auspice route or not.
+ *
+ * SIDE-EFFECTS:
+ * Adds the following properties to `req`, which are used by subsequent middleware.
+ * `req.sendToAuspice` {bool}
+ */
+const isRequestBackedByAuspiceDataset = async (req, res, next) => {
+  // TODO: this is similar to `getDataset` and so we should refactor that
+  // and re-use it here so that we don't let them get out of sync.
+  req.sendToAuspice = false; // explicit default
+  try {
+
+    /* Previous to April 2021 work, _ALL_ these routes would be handled by Auspice.
+    We preserve previous behavior with this block, whilst slowly moving entries out
+    as they are build into gatsby.
+    (e.g. /groups/blab will not be handled by Auspice in the long run) */
+    const pathParts = req.path.replace(/^\//, '').replace(/\/$/, '').split("/");
+    if (
+      pathParts[0]==="status" ||
+      pathParts[0]==="narratives" ||
+      pathParts[0]==="staging" ||
+      pathParts[0]==="groups" ||
+      pathParts[0]==="community" || // note that /community itself is redirected earlier
+      pathParts[0]==="fetch"
+    ) {
+      req.sendToAuspice = true;
+      return next();
+    }
+
+    const {source, prefixParts} = splitPrefixIntoParts(req.path);
+    if (!source.visibleToUser(req.user)) {
+      return next();
+    }
+
+    /* For URLs which look like core builds we know that the start of the path must match a hardcoded list */
+    if (source.name==="core" && !coreBuilds.includes("/"+prefixParts[0])) {
+      return next();
+    }
+
+    /* Extract dataset information & check if it exists, storing the result as `req.sendToAuspice` */
+    const { dataset } = parsePrefix(req.path);
+    if (await datasetExists(dataset)) {
+      req.sendToAuspice = true;
+    }
+    return next();
+  } catch (err) {
+    console.log("isRequestBackedByAuspiceDataset", String(err));
+    req.sendToAuspice = false;
+    return next();
+  }
+};
+
+/**
+ * Does the main dataset file exist?
+ * There is no attempt here to examine the validity or completeness of such a file.
+ * Note: as we don't care about the file contents, there should be a
+ * way to avoid fetching the data here and thus speed up this function.
+ * @returns {bool}
+ */
+async function datasetExists(dataset) {
+  try {
+    if ((await fetch(dataset.urlFor("main"))).status===200) {
```

Comment:
Changing to the [HEAD method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD) can speed up the request since it doesn't return the data.

Reply:
Thanks @joverlee521 -- implementation up at https://github.com/nextstrain/nextstrain.org/pull/325